### PR TITLE
feat(cae): add new datasource to get list of applications

### PR DESCRIPTION
**Pull request by wangyuancheng01**
**What this PR does / why we need it**:
add new datasource to get list of applications.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/cae" TESTARGS="-run TestAccDatasourceApplications_basic"
...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cae -v -run TestAccDatasourceApplications_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceApplications_basic
=== PAUSE TestAccDatasourceApplications_basic
=== CONT  TestAccDatasourceApplications_basic
--- PASS: TestAccDatasourceApplications_basic (18.52s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cae       18.610s
```

### CODE REVIEW

File: docs/data-sources/cae_applications.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the CAE application list.
+  If omitted, the provider-level region will be used.
+
+* `environment_id` - (Required, String) Specifies the ID of the environment.
```
Comment (Lance52259) [inline docs/data-sources/cae_applications.md]:
```suggestion
* `environment_id` - (Required, String) Specifies the ID of the environment to which the applications belong.
```

File: docs/data-sources/cae_applications.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+* `application_id` - (Optional, String) Specifies the ID of the application.
```
Comment (Lance52259) [inline docs/data-sources/cae_applications.md]:
```suggestion
* `application_id` - (Optional, String) Specifies the ID of the application to be queried.
```

File: docs/data-sources/cae_applications.md (hunk 1 above)
Comment (wangyuancheng01) [reply docs/data-sources/cae_applications.md]: done

File: docs/data-sources/cae_applications.md (hunk 2 above)
Comment (wangyuancheng01) [reply docs/data-sources/cae_applications.md]: done

**Pull request merged**